### DISCUSSION TITLE
OpenGL renderer

### DISCUSSION THREAD

**Discussion by ucacaxm** [google/brax#378] [Q&A]
Hi,

I'd like to exploit a model learned in another renderer: GL or even Unity.
I'd like to start writing a OpenGL renderer (not to use it during training) but to visualize the scene and be able to interact with the environment.

Would this be easy to do? Where can I start looking?
I'm looking at image.py but if anyone has a clue ...

Best,
Alex

**Comment by btaba** [comment]
Hi @ucacaxm  , apologies for the late reply. I'd suggest using the simulation/rendering tools in MuJoCo (MJ), they have an OpenGL renderer implemented https://mujoco.readthedocs.io/en/latest/programming/samples.html#sasimulate. Since we use the MuJoCo XML API, the renderers in MJ would maybe fit your use-case?

You'd need to set state from the brax q/qd or qpos/qvel in MJ (https://mujoco.readthedocs.io/en/latest/computation.html?highlight=setstate#the-state). We plan to copy the MJ API more closely going forward so that the interface between brax/MJ is easier to deal with

**Comment by ucacaxm** [comment]
Hi,
Thanks, I'll take a closer look.  This is what I wanted.

>We plan to copy the MJ API more closely going forward so that the interface between brax/MJ is easier to deal with

Great, that will make it easier to switch from one to the other.
